src/schemas.jl

```diff
@@ -440,19 +473,43 @@ function _generate_schema_version_definitions(schema_version::SchemaVersion, par
 end
 
 function _generate_validation_definitions(schema_version::SchemaVersion)
-    field_violation_check_statements = Expr[]
-    for (fname, ftype) in pairs(required_fields(schema_version))
-        fname = Base.Meta.quot(fname)
-        push!(field_violation_check_statements, quote
-            S = $Legolas.accepted_field_type(sv, $ftype)
-            result = $Legolas._check_for_expected_field(ts, $fname, S)
-            isnothing(result) || return $fname => result
+    # When `fail_fast == true`, return first violation found rather than all violations
+    _violation_check = (; fail_fast::Bool) -> begin
+        statements = Expr[]
+        violations = gensym()
+        if !fail_fast
+            push!(statements, :($violations = Pair{Symbol,Union{Type,Missing}}[]))
+        end
+        for (fname, ftype) in pairs(required_fields(schema_version))
+            fname = Base.Meta.quot(fname)
+            push!(statements, quote
+                S = $Legolas.accepted_field_type(sv, $ftype)
+                result = $Legolas._check_for_expected_field(ts, $fname, S)
+            end)
+            push!(statements, if fail_fast
+                :(isnothing(result) || return $fname => result)
+            else
+                :(isnothing(result) || push!($violations, $fname => result))
+            end)
+        end
+        push!(statements, if fail_fast
+            :(return nothing)
+        else
+            :(return $violations)
         end)
+        return statements
+    end
+        return violations
     end
+
```

Comment:
```suggestion
    # When `fail_fast == true`, return first violation found rather than all violations
    _violation_check = (; fail_fast::Bool) -> begin
        statements = Expr[]
        violations = gensym()
        fail_fast || push!(statements, :($violations = Pair{Symbol,Union{Type,Missing}}[]))
        for (fname, ftype) in pairs(required_fields(schema_version))
            fname = Base.Meta.quot(fname)
            found = :($fname => result)
            handle_found = fail_fast ? :(return $found) : :(push!($violations, $found))
            push!(statements, quote
                S = $Legolas.accepted_field_type(sv, $ftype)
                result = $Legolas._check_for_expected_field(ts, $fname, S)
                isnothing(result) || $handle_found
            end)
        end
        push!(statements, :(return $(fail_fast ? :nothing : violations)))
        return statements
    end
```